### ISSUE TITLE
Autocompletion for string literals can contain duplicates

### ISSUE THREAD

**Issue by hediet** [Microsoft/TypeScript#15622]
TypeScript version: 2.3 (Playground)

**Code**
```ts
function foo(x: "bla"): void;
function foo(x: "bla"): void;
function foo(x: string) {}
foo("|");
```

**Expected behavior:**
At position `|`, autocomplete should suggest `bla` once.

**Actual behavior:**
Autocompletion suggests `bla` twice:
![image](https://cloud.githubusercontent.com/assets/2931520/25768185/c6bf1058-3200-11e7-9459-ac25e28199c7.png)

**Relevance**
In the example above, duplication might not be a big deal. But when having multiple overloadings (maybe some with `this` restrictions to select a return type) or many items to complete, autocomplete gets messy:
![image](https://cloud.githubusercontent.com/assets/2931520/25768215/0dd9f6b0-3201-11e7-98db-667c01ec2456.png)

**Comment by mhegazy** [comment]
A PR would be welcomed

**Comment by charlespierce** [comment]
PR Opened to fix this: #15754

**Comment by mhegazy** [comment]
thanks @charlespierce !
